dali/python/backend_impl.cc

```diff
@@ -596,11 +596,34 @@ std::unique_ptr<Tensor<Backend> > TensorListGetItemImpl(TensorList<Backend> &t,
 template <typename Backend>
 std::shared_ptr<TensorList<Backend>> TensorListFromListOfTensors(py::list &list_of_tensors,
                                                                  string &layout) {
+  if (list_of_tensors.empty()) {
+    throw std::runtime_error("Cannot create TensorList from an empty list.");
+  }
+
   auto tl = std::make_shared<TensorList<Backend>>(list_of_tensors.size());
   TensorVector<Backend> tv(list_of_tensors.size());
+  int expected_type = -2;
```

Comment:
Why DALI_NO_TYPE ie. `-1` wouldn't work here?

Reply:
It seems like a better (cleaner) choice, but I think that it is theoretically possible for a Tensor to be of type `DALI_NO_TYPE`, but maybe I am overcomplicating things.